beta_test.go

```diff
@@ -0,0 +1,221 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package mathext_test
+
+import (
+	"math"
+	"testing"
+
+	"github.com/gonum/mathext"
+)
+
+func fequal(a, b float64) bool {
+	return (a - b) < 1e-12
+}
+
+var betaTests = []struct {
+	p, q float64
+	want float64
+}{
+	{
+		p:    1,
+		q:    2,
+		want: 0.5, // obtained from scipy.special.beta(1,2) (version=0.18.0)
+	},
+	{
+		p:    10,
+		q:    20,
+		want: 4.9925087406346778e-09, // obtained from scipy.special.beta(10,20) (version=0.18.0)
+	},
+	{
+		p:    +0,
+		q:    10,
+		want: math.Inf(+1),
+	},
+	{
+		p:    -0,
+		q:    10,
+		want: math.Inf(+1),
+	},
+	{
+		p:    0,
+		q:    0,
+		want: math.NaN(),
+	},
+	{
+		p:    0,
+		q:    math.Inf(-1),
+		want: math.NaN(),
+	},
+	{
+		p:    10,
+		q:    math.Inf(-1),
+		want: math.NaN(),
+	},
+	{
+		p:    0,
+		q:    math.Inf(+1),
+		want: math.NaN(),
+	},
+	{
+		p:    10,
+		q:    math.Inf(+1),
+		want: math.NaN(),
+	},
+	{
+		p:    10,
+		q:    math.NaN(),
+		want: math.NaN(),
+	},
+	{
+		p:    math.NaN(),
+		q:    10,
+		want: math.NaN(),
+	},
+	{
+		p:    math.NaN(),
+		q:    0,
+		want: math.NaN(),
+	},
+	{
+		p:    -1,
+		q:    0,
+		want: math.NaN(),
+	},
+	{
+		p:    -1,
+		q:    +1,
+		want: math.NaN(),
+	},
+	{
+		p:    0,
+		q:    -1,
+		want: math.NaN(),
+	},
+}
+
+func TestBeta(t *testing.T) {
+	for i, test := range betaTests {
+		v := mathext.Beta(test.p, test.q)
+		var equal func(x float64) bool
```

Comment:
Just do
`if math.Abs(v - test.want) > 1e-14 {
    if math.IsNaN(v) && math.IsNaN(test.want) {
       continue
    }`

No need for the function wrapping.